opus/application/static_media/js/widgets.js

```diff
@@ -85,17 +111,249 @@ var o_widgets = {
             let minVal = $(e.currentTarget).data("min");
             let maxVal = $(e.currentTarget).data("max");
             let widgetId = $(e.currentTarget).data("widget");
+            let minInputSlug = $(e.currentTarget).parent(".container").attr("data-mininput");
 
             // NOTE: We need support both RANGE & STRING inputs, for now we implement RANGE first.
             if ($(`#${widgetId} input.RANGE`).length !== 0) {
-                o_widgets.fillRangesInputs(widgetId, maxVal, minVal);
+                o_widgets.fillRangesInputs(widgetId, minInputSlug, maxVal, minVal);
                 // close dropdown and trigger the search
                 $(`#${widgetId} input.op-range-input-min`).dropdown("toggle");
                 $(`#${widgetId} input.RANGE`).trigger("change");
             }
         });
 
         o_widgets.addPreprogrammedRangesBehaviors();
+        o_widgets.addAttachOrRemoveInputsBehaviors();
+    },
+
+    attachAddInputIcon: function(slug, addInputIcon) {
+        /**
+         * Make sure "+ (OR)" is attached to the right of last input set.
+         * Display or hide the icon based on the number of input sets.
+         */
+        $(`#widget__${slug} .op-search-inputs-set`).last().append(addInputIcon);
+        let numberOfInputSets = $(`#widget__${slug} .op-search-inputs-set`).length;
+        if (numberOfInputSets === opus.maxAllowedInputSets) {
+            $(`#widget__${slug} .op-add-inputs`).addClass("op-hide-element");
+        } else {
+            $(`#widget__${slug} .op-add-inputs`).removeClass("op-hide-element");
+        }
+    },
+
+    addAttachOrRemoveInputsBehaviors: function() {
+        /**
+         * Add event handlers for click events on "+ (OR)" and trash icons.
+         * These are behaviors of adding/removing inputs in a single widget.
+         */
+
+        // Create a new set of inputs when clicking the "+ (OR)" button in a widget.
+        $("#search").on("click", ".op-add-inputs-btn", function(e) {
+            e.preventDefault();
+            // When normalize input api is in progress, we have to disable "+ (OR)" so
+            // that no renumber will not happen. This will make sure the corresponding
```

Comment:
"no renumber will not happen" is redundant.
Also isn't it OK for +OR to happen? There's no renumber in this case and it would solve the problem I mentioned earlier.